recipes/zephyr-7b-beta/sft/config_qlora.yaml

```diff
@@ -1,18 +1,22 @@
 # Model arguments
 model_name_or_path: mistralai/Mistral-7B-v0.1
-torch_dtype: auto
-use_flash_attention_2: true
+model_revision: main
+torch_dtype: float16
```

Comment:
Is it correct that training was done with `float16` for the qlora training but `bfloat16` for full parameter training? (And any reason for this, if so?)

Reply:
Yes that's correct - the main reason for these dtypes is that with 4-bit quantization, the other modules will be cast to `float16` by default and I prefer to be explicit about this, while `bfloat16` is needed for compatibility with FlashAttention2

Reply:
Thanks @lewtun ! 

>  main reason for these dtypes is that with 4-bit quantization, the other modules will be cast to float16 by default 

I assume you are referring to `bnb_4bit_compute_dtype` being set to `bfloat16`. Is there merit in making this configurable? 

Since the mistral 7b base is `bfloat16` by default and the SFT step is done in `bfloat16`, would having a consistent type by also setting `compute_dtype` to `bfloat16` (and `torch_dtype` to `bfloat16`) have any benefit? 

I'm no expert here (on the theory or on how BNB works) - my assumption is just that since they have different dynamic ranges there may be some benefit to remaining consistent with the compute dtype.

Reply:
> I assume you are referring to bnb_4bit_compute_dtype being set to bfloat16 in get_quantization_config. Is there merit in making this configurable?

Ah yes, I'm referring to this [line](https://github.com/huggingface/alignment-handbook/blob/c69ae4b8a5436885b72475914b92dfb341876bf0/src/alignment/model_utils.py#L44) and I think it would be good to set this as whatever the `torch_dtype` is in `model_args` (with `float16` the default)

> Since the mistral 7b base is bfloat16 by default, would having a consistent type by also setting compute_dtype to bfloat16 (and torch_dtype to bfloat16) have any benefit?

We haven't tested the effect of `bfloat16` vs `float16` with QLoRA, so once this PR is merged I can run a few experiments to test :)